src/library/vm/vm.h

```diff
@@ -59,6 +59,12 @@ class vm_obj_cell {
 #define lean_vm_check(cond) { if (LEAN_UNLIKELY(!(cond))) vm_check_failed(#cond); }
 #endif
 
+/** Hashes a VM object. If it's a `vm_external`, then
+ * use `vm_external::hash()`.
+ * [warning] `hash()` is not implemented for all of the inheritors of `vm_external`
+ * and by default returns zero. So this means that if the hashes are the same
```

Comment:
Hash collisions are a fact of life, I don't think you need to explicitly mention them here.